sdk/search/Azure.AI.Search/src/SearchServiceClient.cs

```diff
@@ -0,0 +1,246 @@
+                Version.ToVersionString());
+        }
+
+        /// <summary>
+        /// Get a SearchIndexClient for the given <paramref name="indexName"/>
+        /// to use for document operations like querying or adding documents to
+        /// a Search Index.
+        /// </summary>
+        /// <param name="indexName">
+        /// The name of the desired Search Index.
+        /// </param>
+        /// <returns>
+        /// A SearchIndexClient for the desired Search Index.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when the <paramref name="indexName"/> is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the <paramref name="indexName"/> is empty.
+        /// </exception>
+        /// <remarks>
+        /// The same request <see cref="HttpPipeline"/> (including
+        /// authentication and any other configuration) will be used for the
+        /// SearchIndexClient.
+        /// </remarks>
+        public virtual SearchIndexClient GetSearchIndexClient(string indexName)
+        {
+            Argument.AssertNotNullOrEmpty(indexName, nameof(indexName));
+            return new SearchIndexClient(
+                Endpoint,
+                indexName,
+                Pipeline,
+                ClientDiagnostics,
+                Version);
+        }
+
+        /// <summary>
+        /// Gets service level statistics for a Search Service.
+        ///
+        /// This operation returns the number and type of objects in your
+        /// service, the maximum allowed for each object type given the service
+        /// tier, actual and maximum storage, and other limits that vary by
+        /// tier.  This request pulls information from the service so that you
+        /// don't have to look up or calculate service limits.
+        ///
+        /// Statistics on document count and storage size are collected every
+        /// few minutes, not in real time.  Therefore, the statistics returned
+        /// by this API may not reflect changes caused by recent indexing
+        /// operations.
+        /// </summary>
+        /// <param name="clientRequestId">
+        /// An optional caller-defined value that identifies the given request.
+        /// If specified, this will be included in response information as a
+        /// way to map the request.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>The service level statistics.</returns>
+        /// <exception cref="RequestFailedException">
+        /// Thrown when a failure is returned by the Search Service.
+        /// </exception>
+        [ForwardsClientCalls]
+        public virtual Response<SearchServiceStatistics> GetStatistics(
+            Guid? clientRequestId = null,
```

Comment:
Does this really rise to the level of being its own parameter? In every method (for consistency)? What about an "options bag" instead?

Reply:
We'll be switching to something like `RequestParameters` in the near future.